app/src/main/java/ch/epfl/sdp/appart/AdCreationActivity.java

```diff
@@ -257,4 +261,12 @@ private void fillHorizontalViewWithPictures(LinearLayout horizontalLayout, List<
         }
     }
 
+    protected void setPanoramas(List<Uri> uris) {
+        this.panoramaUris = uris;
+    }
+
+    protected void setPictures(List<Uri> uris) {
+        this.picturesUris = uris;
+    }
+
```

Comment:
Unfortunately because I was using them in the `AdCreationDemoActivity`. If you have a better design idea, I'm all ears for it.

Reply:
Instead of creating two activities you could try to use fragments that allow you to share the same viewmodel, but since for the moment we are working completely only with the activities we can also keep this solution and then think about a future refactor.